src/fidesops/service/masking/strategy/masking_strategy_factory.py

```diff
@@ -10,70 +10,56 @@
-    aes_encrypt = AesEncryptionMaskingStrategy
-    hmac = HmacMaskingStrategy
+class MaskingStrategyFactory:
+    registry: Dict[str, MaskingStrategy] = {}
+    valid_strategies = ""
 
     @classmethod
-    def __contains__(cls, item: str) -> bool:
-        try:
-            cls[item]
-        except KeyError:
-            return False
+    def register(cls, name: str) -> Callable[[MaskingStrategy], MaskingStrategy]:
+        def wrapper(strategy_class: MaskingStrategy) -> MaskingStrategy:
+            logger.debug(
+                f"Registering new masking strategy '{strategy_class}' under name '{name}'"
+            )
 
-        return True
+            if name in cls.registry:
+                logger.warning(
+                    f"Masking strategy with name '{name}' already exists. It previously referred to class '{cls.registry[name]}', but will now refer to '{strategy_class}'"
+                )
 
+            cls.registry[name] = strategy_class
+            cls.valid_strategies = ", ".join(cls.registry.keys())
+            return cls.registry[name]
 
-def get_strategy(
-    strategy_name: str,
-    configuration: Dict[
-        str,
-        Union[str, FormatPreservationConfig],
-    ],
-) -> MaskingStrategy:
-    """
-    Returns the strategy given the name and configuration.
-    Raises NoSuchStrategyException if the strategy does not exist
-    """
-    if not SupportedMaskingStrategies.__contains__(strategy_name):
-        valid_strategies = ", ".join([s.name for s in SupportedMaskingStrategies])
-        raise NoSuchStrategyException(
-            f"Strategy '{strategy_name}' does not exist. Valid strategies are [{valid_strategies}]"
-        )
-    strategy = SupportedMaskingStrategies[strategy_name].value
-    try:
-        strategy_config: MaskingConfiguration = strategy.get_configuration_model()(
-            **configuration
-        )
-        return strategy(configuration=strategy_config)
-    except ValidationError as e:
-        raise FidesopsValidationError(message=str(e))
+        return wrapper
 
+    @classmethod
+    def get_strategy(
+        cls,
+        strategy_name: str,
+        configuration: Dict[str, Union[str, FormatPreservationConfig]],
+    ) -> MaskingStrategy:
+        """
+        Returns the strategy given the name and configuration.
+        Raises NoSuchStrategyException if the strategy does not exist
+        """
+        if strategy_name not in cls.registry:
+            raise NoSuchStrategyException(
+                f"Strategy '{strategy_name}' does not exist. Valid strategies are [{cls.valid_strategies}]"
+            )
+        strategy: MaskingStrategy = cls.registry[strategy_name]
+        try:
+            strategy_config: MaskingConfiguration = strategy.get_configuration_model()(
+                **configuration
+            )
+            return strategy(configuration=strategy_config)
+        except ValidationError as e:
+            raise FidesopsValidationError(message=str(e))
```

Comment:
We gain a little clarity and lose an unnecessary `MaskingConfiguration` import with this refactor:
```suggestion
        """

        try:
            strategy = cls.registry[strategy_name]
        except KeyError:
            raise NoSuchStrategyException(
                f"Strategy '{strategy_name}' does not exist. Valid strategies are [{cls.valid_strategies}]"
            )

        try:
            strategy_config = strategy.get_configuration_model()(**configuration)
        except ValidationError as e:
            raise FidesopsValidationError(message=str(e))

        return strategy(configuration=strategy_config)
```